### PR TITLE
ImageList Component

### PR DESCRIPTION
**Pull request by shiftsave**
Take a look at how they're looking in [http://localhost:3000/patterns](http://localhost:3000/patterns) once the app launches. Still very WIP but gets you an idea of where my head is at.

#### To-do

- [x] Extend ImageList component via props to accommodate styles for all three expected views of the component
- [x] Move `ImageList` component to `/components/baseline`
- [x] Add text area
- [x] Style text area
- [x] Connect text area to the database 
- [x] Implement styles for `reference/moodboard` image list
- [x] Implement styles for `storyboards` image list
- [x] Implement styles for `styleframes` image list
- [x] Add ability to delete images from collection and permanently delete them from dropbox if they are unused.
- [x] Connect tile-size to database
- [x] Create a full-screen image viewer

### CODE REVIEW

File: src/components/baseline/components/Card.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
-export const Card = ({ children }) => (
-  <div className="Card">
-    {children}
-  </div>
-);
+export const Card = ({ children, className }) => {
+
+  const styles = classNames({
+    Card: true,
+    [className]: className
```
Comment (jonathanhudak) [inline src/components/baseline/components/Card.js:8]: you can shorten this with just simply `className` once. Its a shorthand

File: src/routes/patterns/components/Buttons.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import { Button, ButtonGroup, Card } from 'components/baseline';
 
 const Buttons = () => (
-  <Card>
+  <Card className='Buttons'>
```
Comment (jonathanhudak) [inline src/routes/patterns/components/Buttons.js:6]: we should get consistent about quotes on html attributes. They should all have double quotes. Maybe adding eslint rules would help

File: src/routes/project/components/Moodboard.js [hunk 3]
```diff
@@ -25,9 +24,13 @@ class MoodboardViewer extends Component {
     const path = this.props.projectPath;
     const collectionId = MOODBOARD;
     const images = this.props.getCollectionFiles({ path, collectionId });
+
     const list = images && images.length
       ? <ImageList content={images} />
       : null;
+
+    console.log(list)
```
Comment (jonathanhudak) [inline src/routes/project/components/Moodboard.js]: remove

**Review by jonathanhudak — APPROVED** [review]
just small suggestions. looks good!

File: src/routes/patterns/components/Buttons.js (hunk 2 above)
Comment (shiftsave) [reply src/routes/patterns/components/Buttons.js:6]: @hudakdidit we should implement prettier

Comment (jonathanhudak) [reply src/routes/patterns/components/Buttons.js:6]: I have it installed in my text editor. I've been using it already, but what your saying is we should enforce it?

Comment (shiftsave) [reply src/routes/patterns/components/Buttons.js:6]: yeah, as part of the build process, we have it setup to run on `git commit` at work.

File: src/components/baseline/components/Card.js (hunk 1 above)
Comment (shiftsave) [reply src/components/baseline/components/Card.js:8]: 👍

File: src/routes/project/components/Moodboard.js (hunk 3 above)
Comment (shiftsave) [reply src/routes/project/components/Moodboard.js]: 👍

File: src/components/baseline/components/ImageList.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    this.setState({ items: content });
+  }
+
+  handleClick = selection => this.setState({ selection, showViewer: true });
+
+  onSortEnd = ({ oldIndex, newIndex }) =>
+    this.props.onReorder(arrayMove(this.state.items, oldIndex, newIndex));
+
+  shouldCancelStart = e => {
+    const disabledElements = [
```
Comment (jonathanhudak) [inline src/components/baseline/components/ImageList.js]: need to move this into a utility to use with `react-sortable-hoc` this is a more robust solution. Or should submit a PR to `react-sortable-hoc` to take a list of class names to disable dnd behavior as needed.

File: src/constants/file.js [hunk 5]
```diff
@@ -3,3 +3,5 @@ export const DELETE_FILE = 'DELETE_FILE';
 export const ADD_FILE_TO_COLLECTION = 'ADD_FILE_TO_COLLECTION';
 export const REMOVE_FILE_FROM_COLLECTION = 'REMOVE_FILE_FROM_COLLECTION';
 export const FILE_DATABASE_DIRECTORY = '_____DB_____';
+export const UPDATE_COLLECTION_ORDER = 'UPDATE_COLLECTION_ORDER';
```
Comment (jonathanhudak) [inline src/constants/file.js:6]: remove

File: src/containers/FileManager.js [hunk 6]
```diff
@@ -32,8 +33,13 @@ export default function FileManager(Component) {
         console.log('gotta delete the folder agian');
         return;
       }
-      this.props.dispatch(addFile(file, path));
-      this.props.dispatch(addFileToCollection(file.name, path, collectionId));
+      if (this.props.files.get("archive").has(file.name)) {
+        // TODO: prevent double upload to dropbox if file name matches
```
Comment (jonathanhudak) [inline src/containers/FileManager.js]: open up a bug and link

File: src/actions/index.js [hunk 7]
```diff
@@ -75,3 +75,17 @@ export const removeFileFromCollection = (id, collectionKey) => {
...
   };
 }
+export const updateCollection = (collectionKey, collection) => {
+  return {
+    type: UPDATE_COLLECTION,
+    collectionKey,
+    collection
+  }
+}
+export const updateCollectionOrder = (collectionKey, collection) => {
```
Comment (jonathanhudak) [inline src/actions/index.js]: remove this action

**Comment by jonathanhudak** [comment]
@shiftsave added image caption persistence in db. It may be buggy on a brand new project, could you try and let me know what you encounter?

**Comment by shiftsave** [comment]
@hudakdidit I just tried it, it saving to the database works, however the comment does not move with the image on re-order, it remains at the same location. I just pushed a few changes too, just started a new file for form elements in the reusable library.

File: src/reducers/files.js [hunk 8]
```diff
@@ -30,16 +30,16 @@ const archive = (state = initialState.get("archive"), action) => {
 
 const collection = (state = List([]), action) => {
   const { id } = action;
-  const collectionEntry = { id };
+  const collectionEntry = Map({ id });
```
Comment (jonathanhudak) [inline src/reducers/files.js]: @shiftsave this is a breaking change. You may need to wipe out your database so it can be re-written

Comment (shiftsave) [reply src/reducers/files.js]: @hudakdidit 👏 👏 this works so great!

**Comment by shiftsave** [comment]
@hudakdidit I added a 2/2 button grid to function as an image size toggle for references images but don't know how to connect it so that the data can be set and persist in the database.

The buttons should map to the following classes:
1sq = base
2sq (horizontal) -  long
2sq (vertical) - tall
4sq - full

File: src/components/baseline/components/Button.js [hunk 9]
```diff
@@ -23,7 +23,7 @@ export const Button = props => {
 
   const styles = classNames({
     Button: true,
-    className,
+    [className]: className,
```
Comment (jonathanhudak) [inline src/components/baseline/components/Button.js]: just `className` will do :)

Comment (jonathanhudak) [reply src/components/baseline/components/Button.js]: or was there a reason to change this back?

File: src/routes/patterns/components/ImageLists.js [hunk 10]
```diff
@@ -1,13 +1,13 @@
...
+import { Button, ImageList } from "components/baseline";
 
 const references = [
-  { url: require("./content/references/01.jpg"), name: "1" },
-  { url: require("./content/references/02.png"), name: "2" },
-  { url: require("./content/references/03.jpg"), name: "3" },
-  { url: require("./content/references/05.jpg"), name: "5" },
-  { url: require("./content/references/06.jpg"), name: "6" },
-  { url: require("./content/references/04.jpg"), name: "4" }
+  { url: require("./content/references/01.jpg"), name: "1", size: "long" },
```
Comment (jonathanhudak) [inline src/routes/patterns/components/ImageLists.js]: @shiftsave this property conflicts with one we already have when we get the images from Dropbox. They have the size property already which is bytes. Maybe we should name this `aspectRatio`?

File: src/components/baseline/components/Button.js (hunk 9 above)
Comment (shiftsave) [reply src/components/baseline/components/Button.js]: Just className sets it it "className" not its actual value

**Pull request merged**